File: lib/scuti/model/host.ex
# Copyright 2023 Clivern. All rights reserved.
# Use of this source code is governed by the MIT
# license that can be found in the LICENSE file.

defmodule Scuti.Model.Host do
  @moduledoc """
  Host Model
  """

  use Ecto.Schema
  import Ecto.Changeset

  schema "hosts" do
    field :uuid, Ecto.UUID
    field :host_group_id, :id
    field :name, :string
    field :hostname, :string
    field :private_ips, :string
    field :public_ips, :string
    field :labels, :string
    field :status, :string
    field :secret_key, :string
    field :reported_at, :utc_datetime

    timestamps()
  end

  @doc false
  def changeset(host, attrs) do
    host
    |> cast(attrs, [
      :uuid,
      :host_group_id,
      :name,
      :hostname,
      :private_ips,
      :public_ips,
      :labels,
      :status,
      :secret_key,
      :reported_at
    ])
    |> validate_required([
      :uuid,
      :host_group_id,
      :name,
      :hostname,
      :private_ips,
      :public_ips,
      :labels,
      :status,
      :secret_key,
      :reported_at
    ])
  end
end

File: agent/core/model/host.go
// Copyright 2022 Clivern. All rights reserved.
// Use of this source code is governed by the MIT
// license that can be found in the LICENSE file.

package model

import (
	"time"

	"github.com/clivern/mandrill/core/util"
)

// Host struct
type Host struct {
	ID         int       `json:"id"`
	TID        int       `json:"tid"`
	HGID       int       `json:"hgid"`
	UUID       string    `json:"uuid"`
	Name       string    `json:"name"`
	HostName   string    `json:"hostName"`
	PrivateIPs string    `json:"privateIPs"`
	PublicIPs  string    `json:"publicIPs"`
	Labels     string    `json:"labels"`
	Status     string    `json:"status"`
	Meta       string    `json:"meta"`
	ReportedAt time.Time `json:"reportedAt"`
	CreatedAt  time.Time `json:"createdAt"`
	UpdatedAt  time.Time `json:"updatedAt"`
}

// Hosts struct
type Hosts struct {
	Hosts []Host `json:"hosts"`
}

// LoadFromJSON update object from json
func (h *Host) LoadFromJSON(data []byte) error {
	return util.LoadFromJSON(h, data)
}

// ConvertToJSON convert object to json
func (h *Host) ConvertToJSON() (string, error) {
	return util.ConvertToJSON(h)
}

// LoadFromJSON update object from json
func (h *Hosts) LoadFromJSON(data []byte) error {
	return util.LoadFromJSON(h, data)
}

// ConvertToJSON convert object to json
func (h *Hosts) ConvertToJSON() (string, error) {
	return util.ConvertToJSON(h)
}

// IsNil if option is nil
func (h *Host) IsNil() bool {
	return h.ID == 0
}
